R/utils.R

```diff
@@ -52,3 +52,18 @@ transformed_distributions <- list(
   "weibull" = list(trans = function(y) log(y), itrans = function(y) exp(y), dist = 'extreme_value'),
   "exponential" = list(trans = function(y) log(y), itrans = function(y) exp(y), dist = 'extreme_value')
 )
+
+#' @title Check if completely left or right censored
+#'
+#' @description
+#' Check if the target matrix is either completely left censored or completely 
+#' right censored and print an error message accordingly.
+#'
+#' @param y The input feature matrix.
+check_censorship <- function(y)
+{
+  if(all(is.na(y[,1])) || all(is.infinite(y[,1])) && !survival::is.Surv(y))
```

Comment:
what if someone passes a mix of `na`s and `inf`s?

Reply:
Ohh yeah.
So I'll fix this by turning the `inf` values to `nan` in `iregnet.R` itself. Previously, it was being done in `get_censoring_type` in `iregnet_fit.cpp`
Done.